eventuate-tram-examples-common/src/main/java/io/eventuate/tram/examples/basic/commands/AbstractTramCommandTest.java

```diff
@@ -0,0 +1,70 @@
+package io.eventuate.tram.examples.basic.commands;
+
+import io.eventuate.tram.commands.common.ReplyMessageHeaders;
+import io.eventuate.tram.commands.producer.CommandProducer;
+import io.eventuate.tram.messaging.common.Message;
+import io.eventuate.tram.messaging.consumer.MessageConsumer;
+import org.junit.jupiter.api.Test;
+import org.springframework.transaction.TransactionDefinition;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public abstract class AbstractTramCommandTest {
+
+  @Inject
+  private CommandProducer commandProducer;
+
+  @Inject
+  private AbstractTramCommandTestConfig config;
+
+  @Inject
+  private MessageConsumer messageConsumer;
+
+  @Inject
+  private TransactionTemplate transactionTemplate;
+
+  private BlockingQueue<Message> queue = new LinkedBlockingDeque<>();
+
+  @Test
+  public void shouldInvokeCommand() throws InterruptedException {
+
+    subscribeToReplyChannel();
+
+    transactionTemplate.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
```

Comment:
Why repeat this in every test? It would be better to either (a) inject a bean that's already initialized or (2) write a `executeInTransaction()` helper method (perhaps on a helper bean that just does everything

Reply:
I will fix, thank you